File: src/main/resources/static/js/service/noticeService.js
//服务层
app.service('noticeService', function ($http) {

    //读取列表数据绑定到表单中
    this.findAll = function () {
        return $http.get('../notice/findAll.do');
    }
    //分页
    this.findPage = function (page, rows) {
        return $http.get('../notice/findPage.do?page=' + page + '&rows=' + rows);
    }
    //通过公告id查询一个公告信息
    this.findOne = function (id) {
        return $http.get('/notice/findOne/' + id);
    };
    //增加公告信息
    this.add = function (loginStuCode, entity) {
        return $http.post('/notice/add/' + loginStuCode, entity);
    };
    //修改公告信息
    this.update = function (stuCode,entity) {
        return $http.post('/notice/update/'+stuCode, entity);
    };
    //删除公告信息
    this.dele = function (notId) {
        return $http.delete('/notice/delete/' + notId);
    };
    //分页查询公告信息
    this.search = function (page, rows, stuCode, role, searchEntity) {
        return $http.post('/notice/search/' + page + '/' + rows + '/' + stuCode + '/' + role, searchEntity);
    }
});

File: src/main/java/com/ccut/yiyi/service/NoticeService.java
package com.ccut.yiyi.service;

import com.ccut.yiyi.model.Notice;
import com.ccut.yiyi.model.group.NoticeGroup;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.Map;

/**
 * @ClassName: NoticeService    公告服务层接口
 * @Description: TODO
 * @Author: Mark
 * @Date: 2020/2/15 20:11
 * @version: V1.0
 */
public interface NoticeService {
    /**
     * 发布社团公告
     *
     * @param stuCode 发布公告的管理员学号
     * @param notice  公告内容
     */
    void add(String stuCode, Notice notice);

    /**
     * 条件查询+分页
     *
     * @param searchMap 查询条件
     * @param page      页码
     * @param rows      每页显示几条
     * @param stuCode   学号
     * @param role      角色
     * @return 返回数据
     */
    Page<NoticeGroup> findSearch(Map searchMap, int page, int rows, String stuCode, String role);

    /**
     * UI页面分页查询公告信息
     *
     * @param page 页码
     * @param rows 条数
     * @return  返回数据
     */
    Page<NoticeGroup> findSearch(int page, int rows);

    /**
     * 通过公告id查询一个公告详情
     *
     * @param id 公告id
     * @return 返回数据
     */
    Notice findOne(Integer id);

    /**
     * 修改公告信息
     *
     * @param stuCode 修改人学号
     * @param notice  修改信息
     */
    void update(String stuCode, Notice notice);

    /**
     * 删除公告信息
     *
     * @param notId 公告id
     */
    void delete(Integer notId);

    /**
     * UI首页查询6个公告信息
     *
     * @return 返回数据
     */
    List<NoticeGroup> findAll();

}
